figures/workflow_montages/scripts/get_psuedo_colored_images.py

```diff
@@ -0,0 +1,443 @@
+df = pd.read_parquet(single_cell_profiles)
+df["Metadata_Well_FOV"] = df["Metadata_Well"] + "_F" + df["Metadata_FOV"]
+df["Metadata_Time"] = df["Metadata_Time"].astype(float).astype(int)
+df["Metadata_Time"].sort_values()
+df.head()
+
+
+# In[5]:
+
+
+list_of_dfs = []
+# check if both are false or true, only one can be true
+if not (composite ^ psuedo_color):
+    raise ValueError("Either composite or psuedo_color must be True, but not both.")
+total = 0
+written = 0
+existsing = 0
+for well_fov in tqdm(
+    df["Metadata_Well_FOV"].unique(),
+    desc="Processing well_fov",
+    unit=" well_fov",
+    leave=False,
+):
+    tmp_df = df.loc[(df["Metadata_Well_FOV"] == well_fov)]
+    input_file_parent_path = pathlib.Path(
+        "/home/lippincm/4TB_A/live_cell_timelapse_apoptosis/"
+        "2.cellprofiler_ic_processing/illum_directory/timelapse/"
+        f"20231017ChromaLive_6hr_4ch_MaxIP_{well_fov}"
+    ).resolve(strict=True)
+    for timepoint in tqdm(
+        tmp_df["Metadata_Time"].unique(),
+        desc="Processing timepoint",
+        unit=" timepoint",
+        leave=False,
+    ):
+
+        tmp_time_df = tmp_df.loc[tmp_df["Metadata_Time"] == timepoint].copy()
+        tmp_time_df = tmp_time_df.drop_duplicates(
+            subset=["Metadata_Well_FOV"]
+        )  # Remove inplace=True
+        output_df = generate_image_pannel_whole_image(
+            df=tmp_time_df,
+            input_file_parent_path=input_file_parent_path,
+            composite=composite,
+            psuedo_color=psuedo_color,
+        )
+        if composite:
+            image = output_df["composite"][0]
+            # save path
+            output_save_path = pathlib.Path(
+                f"../data/whole_image_composite_images/{well_fov}_{timepoint:03d}_composite.png"
+            )
+            output_save_path.parent.mkdir(parents=True, exist_ok=True)
+            total += 1
+            if not output_save_path.exists():
+                image.save(output_save_path)
+                written += 1
+            else:
+                existsing += 1
+        if psuedo_color:
+            cyan, magenta, yellow, green = (
+                output_df["cyan"][0],
+                output_df["magenta"][0],
+                output_df["yellow"][0],
+                output_df["green"][0],
+            )
+            # save path
+            output_save_path = pathlib.Path(
+                f"../data/whole_image_psuedocolor_images/{well_fov}_{timepoint:03d}_cyan.png"
+            )
+            total += 4
+            output_save_path.parent.mkdir(parents=True, exist_ok=True)
+            if not output_save_path.exists():
+                cyan.save(output_save_path)
+                written += 1
+            else:
+                existsing += 1
+            output_save_path = pathlib.Path(
+                f"../data/whole_image_psuedocolor_images/{well_fov}_{timepoint:03d}_magenta.png"
+            )
+            if not output_save_path.exists():
```

Comment:
Given how this pattern repeats consider creating a simple utility function, or even a lambda which can perform this check for every save you do here.